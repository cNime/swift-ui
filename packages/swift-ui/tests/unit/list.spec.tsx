import { mount } from '@vue/test-utils'
import List from '../../src/components/list/index'
import ListItem from '../../src/components/list/item/index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(List, {
    components: {
      ListItem,
    },
    slots,
    props: { ...values }
  })
}

describe('List Snap', () => {
  it('render completed List', () => {
    const wrapper = mount({
      data() {
        return {
          iconInfo: {
            size: 25,
            color: '#78A4FA',
            value: 'calendar',
          },
        }
      },
      components: {
        List,
        ListItem,
      },
      template: `
      <List>
        <ListItem title='标题文字' />
        <ListItem title='标题文字' arrow='right' />
        <ListItem title='标题文字' note='描述信息' />
        <ListItem title='禁用状态' disabled extraText='详细信息' />
        <ListItem title='标题文字' note='描述信息' arrow='right' />
        <ListItem title='标题文字' extraText='详细信息' arrow='right' />
        <ListItem
          arrow='right'
          note='描述信息'
          title='标题文字标题文字标题文字标题文字标题文字'
          extraText='详细信息详细信息详细信息详细信息'
        />
        <ListItem
          title='标题文字'
          note='描述信息'
          extraText='详细信息'
          arrow='right'
          thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
        />
        <ListItem
          arrow='right'
          note='描述信息'
          :iconInfo="iconInfo"
          title='标题文字'
          extraText='详细信息'
          thumb='http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png'
        />
        <ListItem title='标题文字' isSwitch />
        <ListItem title='标题文字' isSwitch disabled />
        <ListItem title='标题文字' switchIsCheck isSwitch />
        <ListItem title='标题文字' switchIsCheck isSwitch disabled />
      </List>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render completed List -- no border', () => {
    const wrapper = mount({
      components: {
        List,
        ListItem,
      },
      template: `
      <List :hasBorder="false">
        <ListItem title='标题文字' :hasBorder="false" />
        <ListItem title='标题文字' :hasBorder="false" />
      </List>
      `,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('List Behavior ', () => {
  it('ListItem onClick', () => {
    const onClick = jest.fn()
    const wrapper = mount(ListItem, {
      props: {
        title: 'ListItem title'
      }

    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.emitted()).toHaveProperty('click')
  })

  /* it('ListItem onSwitchChange && onClick', () => {
    const onClick = jest.fn()
    const onChange = jest.fn()
    const wrapper = mount({
      components: {
        List,
        ListItem,
      },
      methods: {
        onClick: onClick,
        onChange: onChange,
      },
      template: `
        <List>
          <ListItem
            title='标题文字'
            is-switch
            :on-click="onClick"
            :on-switch-change="onChange"
          />
        </List>
      `,
    })
    wrapper.find('.at-list .at-list__item .item-extra__switch switch').trigger('tap')
    wrapper.find('.at-list .at-list__item .item-extra__switch switch').trigger('change')
    expect(onClick).toBeCalled()
    expect(onChange).toBeCalled()
  })

  it('ListItem switch was checked', () => {
    const wrapper = mount({
      components: {
        List,
        ListItem,
      },
      template: `
        <List>
          <ListItem
            title='标题文字'
            is-switch
            switch-is-check
          />
        </List>
      `,
    })
    const switchEl = wrapper.find('.at-list .at-list__item .item-extra__switch switch')
    expect(switchEl.attributes('checked')).toBe('checked')
  })

  it('ListItem switch was unchecked', () => {
    const wrapper = mount({
      components: {
        List,
        ListItem,
      },
      template: `
        <List>
          <ListItem
            title='标题文字'
            is-switch
          />
        </List>
      `,
    })
    const switchEl = wrapper.find('.at-list .at-list__item .item-extra__switch switch')
    expect(switchEl.attributes('checked')).toBe(undefined)
  }) */
})
