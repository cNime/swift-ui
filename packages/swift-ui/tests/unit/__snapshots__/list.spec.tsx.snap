// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snap render completed List -- no border 1`] = `
<div
  class="at-list at-list--no-border"
>
  
  <div
    class="at-list__item at-list__item--no-border"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--no-border"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  
</div>
`;

exports[`List Snap render completed List 1`] = `
<div
  class="at-list"
>
  
  <div
    class="at-list__item"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--multiple"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--disabled"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Disabled state
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--multiple"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--multiple"
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title textTitle textTitle textTitle textTitle text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details details details details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--thumb at-list__item--multiple"
  >
    <div
      class="at-list__item-container"
    >
      <div
        class="at-list__item-thumb item-thumb"
      >
        <img
          class="item-thumb__info"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </div>
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--thumb at-list__item--multiple"
  >
    <div
      class="at-list__item-container"
    >
      <div
        class="at-list__item-thumb item-thumb"
      >
        <img
          class="item-thumb__info"
          src="http://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png"
        />
      </div>
      <div
        class="at-list__item-icon item-icon"
      >
        <div
          class="at-icon at-icon-calendar"
          style="color: rgb(120, 164, 250); font-size: 25px;"
        />
      </div>
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          <div
            class="item-content__info-note"
          >
            Description
          </div>
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        <div
          class="item-extra__info"
        >
          details
        </div>
        
        <div
          class="item-extra__icon"
        >
          <div
            class="at-icon item-extra__icon-arrow at-icon-chevron-right"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="at-list__item"
    isswitch=""
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--disabled"
    isswitch=""
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item"
    isswitch=""
    switchischeck=""
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  <div
    class="at-list__item at-list__item--disabled"
    isswitch=""
    switchischeck=""
  >
    <div
      class="at-list__item-container"
    >
      
      
      <div
        class="at-list__item-content item-content"
      >
        <div
          class="item-content__info"
        >
          <div
            class="item-content__info-title"
          >
            Title text
          </div>
          
        </div>
      </div>
      <div
        class="at-list__item-extra item-extra"
      >
        
        
        
      </div>
    </div>
  </div>
  
</div>
`;
